app/controllers/application_controller.rb

```diff
@@ -1,2 +1,64 @@
-class ApplicationController < ActionController::API
+# frozen_string_literal: true
+
+require 'oauth2'
+
+class ApplicationController < ActionController::Base
+  attr_reader :client, :current_user
+
+  include Pundit
+
+  protect_from_forgery with: :null_session
+
+  rescue_from Dgfip::AccessDenied do |e|
+    render status: :unauthorized, json: {
+      message: "Vous n'êtes pas authorisé à accéder à cette API",
+      detail: e.message
+    }
+  end
+
+  rescue_from Pundit::NotAuthorizedError do |_e|
+    render status: :forbidden, json: {
+      message: ["Vous n'êtes pas authorisé à modifier cette ressource"]
+    }
+  end
+
+  rescue_from ActiveRecord::RecordNotFound do |_e|
+    render status: :not_found, json: {
+      message: 'Record not found'
+    }
+  end
+
+  private
+
+  def authenticate!
```

Comment:
Shouldn't all this logic go into a service object?

Reply:
maybe a concern, how does it would look as service ?